File: WebCatalogBackend/src/main/java/com/cafebabe/model/Gpu.java
package com.cafebabe.model;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.Entity;

@Entity
@Getter
@Setter
public class Gpu extends BaseDataObject {

    protected boolean isIntegrated;
    protected GpuType gpuType;
    protected Integer streamingProcessorsAmount;
    protected Integer clockSpeedInMHz;
}

File: WebCatalogFrontend/src/app/model/gpu.ts
import {BaseDataObject} from "./base-data-object";

export class Gpu extends BaseDataObject {
  isIntegrated: boolean;
  gpuType: GpuType;
  streamingProcessorsAmount: number;
  clockSpeedInMHz: number;
}

export enum GpuType {

  MOBILE = 'MOBILE', NOTEBOOK = 'NOTEBOOK', DESKTOP = 'DESKTOP'
}
